test/integration/callbacks_test.rb

```diff
@@ -20,6 +20,8 @@ def test_progress_callback
     assert_equal 1111, progress.last["Tables"]["gftest.test_table_1"]["TargetPaginationKey"]
     assert_equal "completed", progress.last["Tables"]["gftest.test_table_1"]["CurrentAction"]
 
+    assert_equal 0, progress.last["ActiveDataIterators"]
+
```

Comment:
Is it possible that we receive more than one `progress` reports during the ghostferry run? If so, can you add a check that the `ActiveDataIterators` is `>0` in one of the non-last progresses?

Reply:
We do sometimes, so it will introduce flakiness. I've manually verified these feature for now.

we really a better integration testing environment...